T18496

ptomato opened the pull request.

ptomato committed:
commands: Update should not fetch repos with no remote

It's possible to have a git repo with no remote in your checkout dir. If
that's the case, "flapjack update" should not fetch it.

https://phabricator.endlessm.com/T18496

ptomato committed:
commands: Keep going when repo update fails

It's possible for one git repository to fail to update during "flapjack
update" (see previous commit.) If that's the case, print the name of the
repo that failed, otherwise you get no indication which it is; and
continue with the other repos, since that shouldn't be fatal.

https://phabricator.endlessm.com/T18496

ptomato committed:
config: Allow configuring the base runtime branch

You might want to build a dev runtime on top of a stable runtime, e.g.
org.gnome.Sdk//3.26. For that case, you need to be able to configure the
flatpak branch of the SDK, as well as the git branch of the git repo
where the manifest is located.

https://phabricator.endlessm.com/T18496

ptomato committed:
config: Allow operating on flatpak user installation

https://phabricator.endlessm.com/T18496

ptomato committed:
config: Switch to a keyfile configuration

In order to not allow arbitrary code in config files, we switch to an
actual configuration file format. The "configparser" format is built in
to Python's standard library and also is fairly powerful.

https://phabricator.endlessm.com/T18496

cosimoc reviewed (approved):
Didn't do a super thorough review of this, but looks generally pretty good.
@ebassi Do you want to take a look?

ptomato committed:
config: Allow not specifying dev_tools_manifest

We want the default value to be nothing at all here. This requires
allowing a value of None in _string_expandtilde().

https://phabricator.endlessm.com/T18496

ptomato commented:
As it happens I just added another commit to fix up something from my testing today

ebassi reviewed (approved):
I like this very much, thanks!
